src/cargo/sources/registry/remote.rs

```diff
@@ -183,18 +184,27 @@ impl<'cfg> RegistryData for RemoteRegistry<'cfg> {
             Some(blob) => blob,
             None => anyhow::bail!("path `{}` is not a blob in the git repo", path.display()),
         };
-        data(blob.content())
+        Ok(Poll::Ready(data(blob.content())?))
     }
 
     fn config(&mut self) -> CargoResult<Option<RegistryConfig>> {
         debug!("loading config");
         self.prepare()?;
         self.config.assert_package_cache_locked(&self.index_path);
         let mut config = None;
-        self.load(Path::new(""), Path::new("config.json"), &mut |json| {
-            config = Some(serde_json::from_slice(json)?);
-            Ok(())
-        })?;
+        loop {
+            match self.load(Path::new(""), Path::new("config.json"), &mut |json| {
+                config = Some(serde_json::from_slice(json)?);
+                Ok(())
+            })? {
+                Poll::Ready(_) => {
+                    break;
+                }
+                Poll::Pending => {
+                    // TODO: dont hot loop for it to be Ready
```

Comment:
I think this will want the ability to return up `Poll`

Reply:
Happy to do the work, I am just not seeing where up the stack we would want to call this in parallel with other work? What have I missed?

Reply:
Ah that's true, although I think we'll want special handling for this since I don't think we want to add a round-trip time to get this file all the time in Cargo. We'd presumably want to cache this file separately for a fixed length of time and while in that window of time Cargo doesn't re-fetch the file.